File: 05430.cpp
#include <cstdio>
#include <deque>
#include <string>
#include <algorithm>

using namespace std;

int main(){
	int t;
	scanf("%d", &t);
	for(int i = 0; i < t; ++i){
		char command[100001];
		scanf("%s", command);
		int n;
		scanf("%d\n%*c", &n);
		deque<int> deck;
		for(int j = 0; j < n; ++j){
			int x;
			scanf("%d%*c", &x);
			deck.push_back(x);
		}
		bool error = false;
		bool reverse = false;
		for(int j = 0; command[j] != '\0'; ++j){
			if(command[j] == 'R'){
				reverse = !reverse;
			}else if(command[j] == 'D'){
				if(deck.empty()){
					error = true;
					break;
				}else if(!reverse){
					deck.pop_front();
				}else if(reverse){
					deck.pop_back();
				}
			}
		}
		if(error){
			printf("error\n");
			continue;
		}
		printf("[");
		if(!deck.empty() && !reverse){
			printf("%d", deck.front());
			deck.pop_front();
	    	while(!deck.empty()){
	    		printf(",%d", deck.front());
	    		deck.pop_front();
			}
	    }else if(!deck.empty() && reverse){
			printf("%d", deck.back());
			deck.pop_back();
	    	while(!deck.empty()){
	    		printf(",%d", deck.back());
	    		deck.pop_back();
			}
		}
		printf("]\n");
		fflush(stdin);
	}
	return 0;
}

File: 05430.py
def main():
    t = int(input())
    for _ in range(t):
        commands = input()
        n = int(input())
        L = eval(input())
        reverse = False
        first, last = 0, 0
        for command in commands:
            if command == "R":
                reverse = not reverse
            elif command == "D":
                if reverse:
                    last += 1
                else:
                    first += 1
        if first + last > n:
            print("error")
        elif reverse:
            print("[" + ",".join(str(x) for x in L[::-1][last:n-first]) + "]")
        else:
            print("[" + ",".join(str(x) for x in L[first:n-last]) + "]")


main()
